### PR TITLE
Remove cached Type instances

### PR DESCRIPTION
**Pull request by huoyaoyuan**
With #75573, `typeof` is now a constant instead of a call. The workaround to reduce CLR calls should not be necessary now.
Only updating libraries targeting latest .NET version.

There are a lot of instances in `XmlValueConverter`, which is generated by I can't find the generator code.

### CODE REVIEW

**Comment by MichalPetryka** [comment]
This will probably regress Mono perf.

**Comment by hez2010** [comment]
> This will probably regress Mono perf.

While I never think the performance of mono is essential as it will never be used in performance-sensitive scenarios.

**Comment by En3Tho** [comment]
I certainly don't want Blazor to regress though.

**Comment by stephentoub** [comment]
> > This will probably regress Mono perf.
> 
> While I never think the performance of mono is essential as it will never be used in performance-sensitive scenarios.

We do care about not regressing it.

**Comment by huoyaoyuan** [comment]
> This will probably regress Mono perf.

Is it possible to do similar optimization on Mono?

File: src/libraries/System.Private.Xml/src/System/Xml/BinaryXml/XmlBinaryReader.cs [hunk 1]
```diff
@@ -3308,13 +3305,13 @@ private void CheckAllowContent()
...
             map[(int)BinXmlToken.XSD_BINHEX] = TypeOfByteArray;
             map[(int)BinXmlToken.XSD_BASE64] = TypeOfByteArray;
-            map[(int)BinXmlToken.SQL_CHAR] = TypeOfString;
-            map[(int)BinXmlToken.SQL_VARCHAR] = TypeOfString;
-            map[(int)BinXmlToken.SQL_TEXT] = TypeOfString;
-            map[(int)BinXmlToken.SQL_NCHAR] = TypeOfString;
-            map[(int)BinXmlToken.SQL_NVARCHAR] = TypeOfString;
-            map[(int)BinXmlToken.SQL_NTEXT] = TypeOfString;
-            map[(int)BinXmlToken.SQL_UUID] = TypeOfString;
+            map[(int)BinXmlToken.SQL_CHAR] = typeof(string);
```
Comment (jkotas) [inline src/libraries/System.Private.Xml/src/System/Xml/BinaryXml/XmlBinaryReader.cs:3308]: You can keep `TypeOfString` here, just change it to a local - similar to how the other similar TypeOf... in this method are done.

**Review by jkotas — APPROVED** [review]
This does not appear be touching code where the speed is critical for Mono scenarios.

There is a balance between keeping the code small and simple, improving time for CoreCLR, and not regressing time for Mono. We care about all of these - it is not unusual that we have to choose two out of three.

This change makes the code a bit smaller and more amenable to trimming, that are important performance metrics for Mono scenarios as well, so depending on what you measure this is either negligible regression or negligible improvement.

Overall, I think that this is nice cleanup.

**Pull request merged**